lib/services/ngsi/entities-NGSI-v2.js

```diff
@@ -511,6 +535,11 @@ function sendUpdateValueNgsi2(entityName, originMeasures, originTypeInformation,
                         }
                         newAttrMeta['value'] = metaValueExpression;
                         currentAttr.metadata[metaKey] = newAttrMeta;
+
+                        //RE-Populate de JEXLcontext
+                        if (jexlctxt.metadata && jexlctxt.metadata[currentAttr.name]) {
```

Comment:
This `if` clause checks if the metadata exits into the original jexl context but, in the case of new metadatas generated as a result of expressions, it would not be included. I think It should be appended to the jexlctxt even if it was not part previously of it.